kombu/compat.py

```diff
@@ -102,8 +100,8 @@ def __init__(self, connection, queue=None, exchange=None,
 
         self.queue = queue or self.queue
         self.exchange = exchange or self.exchange
-        self.exchange_type = exchange_type or self.exchange_type
-        self.routing_key = routing_key or self.routing_key
+        self.exchange_type = exchange_type or 'direct'
```

Comment:
Is this change an enhancement for readability? This way the default values will be used from the class variables.